src/main.rs

```diff
@@ -250,8 +250,8 @@ pub fn setup_csrs() {
     hie.modify(Interrupt::VirtualSupervisorExternal.to_hie_field().unwrap());
     CSR.hie.set(hie.get());
 
-    // Make counters available to guests.
-    CSR.hcounteren.set(0xffff_ffff_ffff_ffff);
+    // Trap on guest access to counters.
+    CSR.hcounteren.set(0);
```

Comment:
So what about the fixed-function counters? Supervisor software will expect accesses to `time` to "just work" (and legacy software will expect to be able to access `cycle` and `instret` as well, though I suppose we can ignore that for now).

I think we'll need to allow unconditional access to `time`, and for guest VMs use `htimedelta` to do the equivalent of saving/restoring the `time` value when the VM is swapped out. Not sure we should do that for the host VM though; time should probably appear to advance while it's yielded to the TSM or a guest VM. (Though that gets interesting when it comes to nested guest VMs.)

Reply:
Good point...we can enable access to them by default.

Reply:
You want to trap on everything except time. I don't think we have to worry about legacy software with salus.